Add ability to listen on an environment-specified port

MartijnVdS opened the pull request:
The port defaults to "25" as before, but can be overridden by providing the
PORT environment variable on startup.

oba11 commented:
oh cool , please lets remove the line [here](https://github.com/namshi/docker-smtp/blob/master/Dockerfile#L17) and its good to merge.

MartijnVdS commented:
That would require other users to explicitly expose the port using -p 25:25 or similar -- just using docker run -P won't expose it anymore, making the change backward-incompatible in that regard.

MartijnVdS commented:
I've removed the EXPOSE line.

oba11 commented:
sorry about missing that part @MartijnVdS , it would break backward compatibility or anyone using `-P`, please add it back and good to merge. apologies for trouble.

Commit:
Add ability to listen on an environment-specified port

The port defaults to "25" as before, but can be overridden by providing the
PORT environment variable on startup.

MartijnVdS commented:
@oba11 Done!

oba11 commented:
thanks
